Change Lambda implementation and add a helper CLI tool

SivagiriVisakan opened the pull request.

SivagiriVisakan committed:
Split db-migrate wrapper and S3 migration into seperate modules.

SivagiriVisakan committed:
Add a cli tool to invoke the S3-Lambda based migration

SivagiriVisakan committed:
Add example.

senor-coder commented on migration/s3Migrator.js:
```diff
@@ -0,0 +1,82 @@
+ * A class that takes care of executing migrations stored as an
+ * archive(.zip) files in AWS S3
+ */
+export class S3DBMigrator {
```
Can you make it SLSDbMigrator / ServerlessDbMigrator and not S3 ?

senor-coder commented on package.json:
```diff
@@ -14,18 +14,27 @@
     "deploy": "sls deploy"
   },
+  "bin": {
+    "s3migrate": "./bin/s3migrate.js"
```
slsmigrate

SivagiriVisakan committed:
Rename s3migrator to serverlessDbMigrator

SivagiriVisakan committed:
Handle errors when running commands

SivagiriVisakan committed:
Support passing configuration when invoking the lambda.

SivagiriVisakan committed:
Add option to read local config to the CLI.

SivagiriVisakan committed:
Change to commonjs.

SivagiriVisakan committed:
Fix issue with createDB.

SivagiriVisakan committed:
Change command map into a function and remove "sleep".

When the commandMap is used, the "this" binding inside the function
responsible for executing the command is not correct and that causes
issues.

The sleep timeout doesn't seem to be necessary, so removing that.
Incase if there is some error in the future, it maybe needed again.

SivagiriVisakan committed:
bin: Switch to async function and handle errors with non-zero exit code in CLI

SivagiriVisakan committed:
Change serverless.yml and example script

SivagiriVisakan committed:
Add CLI example to README

SivagiriVisakan committed:
Fix typo and change version to v0.1.0
